support/metrics/metricsTracker.go

```diff
@@ -190,34 +188,65 @@ func MakeMetricsTracker(
 	}, nil
 }
 
+// MakeMetricsTrackerGui is a factory method to create a `metrics.Tracker` from the CLI.
+func MakeMetricsTrackerGui(
+	userID string,
+	deviceID string,
+	apiKey string,
+	client *http.Client,
+	botStartTime time.Time,
+	version string,
+	goos string,
+	goarch string,
+	goarm string,
+	guiVersion string,
+) (*MetricsTracker, error) {
+	props := commonProps{
+		CliVersion: version,
+		Goos:       goos,
+		Goarch:     goarch,
+		Goarm:      goarm,
+		GuiVersion: guiVersion,
+	}
+
+	return &MetricsTracker{
+		client:       client,
+		apiKey:       apiKey,
+		userID:       userID,
+		deviceID:     deviceID,
+		props:        props,
```

Comment:
we should set the field `props` to a `map[string]interface{}` on metricsTracker -- we do this once only using the `toMapStringInterface` function and then in mergeProps we can merge two maps instead of doing any json work